### PR TITLE
Version check for the version CLI option.

### PR DESCRIPTION
**Pull request by mundry**
This PR aims to add functionality that runs a version check every time gpx-animator is run with the `--version` option to display the current update state.

It changes the version output as follows.
```
GPX Animator 1.5.0 (A newer version 1.6.1 is available. Visit https://gpx-animator.app/#downloads to update!)
GPX Animator 1.6.1 (latest version)
```

Unfortunately I don't have the means to test on Windows or Mac.

Let me know any and all feedback you have.

### CODE REVIEW

File: src/main/java/app/gpx_animator/ui/cli/CommandLineConfigurationFactory.java [hunk 1]
```diff
@@ -348,6 +361,41 @@ private void normalizeTrimGpxEnd() {
...
+    private static String checkVersion(final ResourceBundle resourceBundle) {
+        final var currentVersion = new DefaultArtifactVersion(Constants.VERSION.replace("-SNAPSHOT", ""));
+
+        try {
+            final var dbf = DocumentBuilderFactory.newInstance();
+            final var db = dbf.newDocumentBuilder();
+            final var doc = db.parse(new URL(Constants.UPDATES_URL).openStream());
+            doc.getDocumentElement().normalize();
+
+            final var entries = doc.getElementsByTagName("entry");
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/ui/cli/CommandLineConfigurationFactory.java:373]: Sorry for my late answer, I am on easter vacation.

This code can be much easier. Every supported OS has always the same version number. So you do not need to check for the OS we are actually running on. It is a nice, working code, but makes this process more complicated than needed.

I would suggest just to load the XML and use a regex to extract the version number from the first occurrence: `newVersion="1.6.1"`

**Review by McPringle — CHANGES_REQUESTED** [review]

File: src/main/resources/i18n/Messages.properties [hunk 2]
```diff
@@ -278,3 +278,7 @@ ui.panel.tracksettings.timeoffset.label=Time Offset
 ui.panel.tracksettings.timepointinterval.label=Forced Point Time Interval
 ui.panel.tracksettings.trimend.label=Trim End of GPX File
 ui.panel.tracksettings.trimstart.label=Trim Start of GPX File
+version.check.error.network=Version check cannot be reached at the moment
+version.check.error.xml=Unknown version data
+version.check.latest=latest version
```
Comment (McPringle) [inline src/main/resources/i18n/Messages.properties:283]: I would like more the term "you are up to date", is this okay for you?

File: src/main/java/app/gpx_animator/ui/cli/CommandLineConfigurationFactory.java [hunk 3]
```diff
@@ -180,9 +188,13 @@ public CommandLineConfigurationFactory(final String... args) throws UserExceptio
                         case ZOOM -> cfg.zoom(Integer.parseInt(args[++i]));
                         case VERSION -> {
                             try (var pw = new PrintWriter(new OutputStreamWriter(System.out, StandardCharsets.UTF_8))) {
-                                pw.println(Constants.APPNAME_VERSION);
+                                pw.print(Constants.APPNAME_VERSION);
+                                pw.print(" (");
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/ui/cli/CommandLineConfigurationFactory.java:192]: I would prefer the output on a new line.

**Review by McPringle — CHANGES_REQUESTED** [review]
Very nice! The code is ready to merge, but I would like two minor text changes. What do you think about it?

File: src/main/java/app/gpx_animator/ui/cli/CommandLineConfigurationFactory.java (hunk 3 above)
Comment (mundry) [reply src/main/java/app/gpx_animator/ui/cli/CommandLineConfigurationFactory.java:192]: Sure, no problem!

**Review by McPringle — APPROVED** [review]

**Comment by McPringle** [comment]
Thank you very much for your work, @mundry, I highly appreciate your contribution to GPX Animator!

**Comment by mundry** [comment]
My pleasure! Thanks for the pleasant experience! :smiley:

**Pull request merged**